onedal/covariance/incremental_covariance.py

```diff
@@ -58,10 +58,48 @@ def __init__(self, method="dense", bias=False, assume_centered=False):
         self._reset()
 
     def _reset(self):
+        self._need_to_finalize = False
         self._partial_result = self._get_backend(
             "covariance", None, "partial_compute_result"
         )
 
+    def __getstate__(self):
+        self.finalize_fit()
+        data = self.__dict__.copy()
+        partial_result_data = dict()
+        partial_result_data["partial_n_rows"] = from_table(
+            data["_partial_result"].partial_n_rows
+        )
+        partial_result_data["partial_crossproduct"] = from_table(
+            data["_partial_result"].partial_crossproduct
+        )
+        partial_result_data["partial_sums"] = from_table(
+            data["_partial_result"].partial_sums
+        )
+        data["_partial_result"] = partial_result_data
```

Comment:
It looks like the partial result object only contains these three attributes: https://github.com/intel/scikit-learn-intelex/blob/main/onedal/covariance/covariance.cpp#L138
```suggestion
        p_res = data["_partial_result"]
        data["_partial_result"] = {key:from_table(value) for key, value in p_res.__dict__.items()}
```

Reply:
since it is not class with dynamic attributes (see https://pybind11.readthedocs.io/en/stable/classes.html) it has no `__dict__` attribute. so the second option is not valid.